Sprint_5

AgonianV opened the pull request.

AgonianV committed:
Добавлены тесты на проверку регистрации.
Выполнен генератор логина

AgonianV committed:
Добавлены тесты на проверку регистрации.
Выполнен генератор логина

AgonianV committed:
Добавлены тесты основных функций сервиса.
Описаны все локаторы используемые в автотестах.

AgonianV committed:
Обновлен gitignore

irisqul commented on .idea/inspectionProfiles/profiles_settings.xml:
Нужно исправить: из финального проекта нужно удалить все файлы и папки, которые не относятся к решению задачи

irisqul commented on locators:
```diff
@@ -0,0 +1,25 @@
+//section[1]/div[1]/div[2] # Кнопка "Соусы"
```
Нужно исправить: согласно заданию нужно использовать локаторы - Определи и опиши локаторы, которые будешь использовать в тестах. Сейчас ты перечисляешь только селекторы, нужно каждый селектор записать в переменную (локатор), и потом вызывать этот локатор в течение проекта.

irisqul commented on locators:
```diff
@@ -0,0 +1,25 @@
+//section[1]/div[1]/div[2] # Кнопка "Соусы"
+//main/section[1]/div[2]/h2[2] # Раздел "Соусы"
```
Нужно исправить: Пожалуйста найди более надежные пути до элементов, идеально если будет указано что-то специальное - вроде текста на кнопке или заголовка или иногда какие то атрибуты содержат отсылку к тому что за элемент. Поиск элементов не должен зависеть от верстки всей страницы

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+
+
+def test_click_on_sauce_section():
+    driver = webdriver.Chrome()
```
Нужно исправить Инстанс драйвера необходимо получать и закрывать в фикстуре

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+    driver.get("https://stellarburgers.nomoreparties.site/")
+
+    #Кликаем на кнопку "Соусы"
+    driver.find_element(By.XPATH, "//section[1]/div[1]/div[2]").click()
```
Нужно исправить: везде по проекту использовать локаторы, селекторы пишем только в файле locators

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+def test_click_on_fillings_section():
+    driver = webdriver.Chrome()
+
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Нужно исправить: вынести все урлы в отдельный файл и переиспользовать их. Тогда, если нужно будет тестировать не продакшн нам нужно будет изменить урлы только в одном файле, а не искать их по всему проекту

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+    section_location = element.location
+
+    # Проверяем, что раздел "соусы" после клика по кнопке переместились наверх
+    assert section_location == {'x': 0, 'y': 244}
```
Нужно исправить: везде - никогда не пишем в ассерте координаты, ориентироваться нужно на элементы по их селекторам

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+
+    # Находим раздел "соусы"
+    element = driver.find_element(By.XPATH, "//main/section[1]/div[2]/h2[2]")
+    time.sleep(2)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,67 @@
+
+    # Проверяем, что раздел "соусы" после клика по кнопке переместились наверх
+    assert section_location == {'x': 0, 'y': 244}
+
```
Нужно исправить: У элемента таба при клике добавляется новый класс. Следует проверять его

irisqul commented on tests/test_registration_page.py:
```diff
@@ -0,0 +1,47 @@
+    driver.find_element(By.XPATH, "//fieldset[1]/div/div/input").send_keys("Denis")
+
+    # Ввод почты в поле Email
+    driver.find_element(By.XPATH, "//fieldset[2]/div/div/input").send_keys(generate_login())
```
Можно лучше: генерировать тестовые данные для сессии в фикстуре

AgonianV committed:
Выполнение правок по пул реквесту.
-добавлены локаторы
-добавлен файл url, где размещены ссылки на страницы
-переписаны пути для кнопок "Соусы", "Начинки", "Булки"
-Убраны слипы, добавлены явные ожидания
-убран файл .idea/**/profiles_settings.xml

AgonianV committed:
Генератор логинов добавлен в фикстуру

AgonianV committed:
Удалил папку .idea из репозитория

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,15 @@
+def driver():
+    driver = webdriver.Chrome()
+
+    return driver
```
Нужно исправить: в этой фикстуре также должно быть завершение работы браузера после тестов

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,56 @@
+
+
+    # Проверяем, что раздел "соусы" после клика по кнопке получили Класс "tab_tab_type_current__2BEPc"
+    assert driver.find_element(By.XPATH, sauce_result).text == "Соусы"
```
а зачем тут теперь проверять текст, текст есть в селекторе. Тут ассерт на наличие такого элемента (isDisplayed)

irisqul commented on tests/test_constuctor_sections.py:
```diff
@@ -0,0 +1,56 @@
+    # Проверяем, что раздел "соусы" после клика по кнопке получили Класс "tab_tab_type_current__2BEPc"
+    assert driver.find_element(By.XPATH, sauce_result).text == "Соусы"
+
+    driver.quit()
```
нужно исправить - из всех тестов эту строку нужно перенести в фиктуру driver

irisqul commented on tests/test_entering_in_personal_account.py:
```diff
@@ -0,0 +1,42 @@
+
+    # Ожидаем пока главная страница не прогрузится
+    WebDriverWait(driver, 5).until(expected_conditions.url_to_be((main_page)))
+
```
Можно лучше: Здесь и далее. Проследи за соблюдением стиля PEP8 в твоем коде, внутри функций - отступы не делаются, между методами одного класса - отступ в 1 строку

irisqul commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,125 @@
+    WebDriverWait(driver, 5).until(expected_conditions.url_to_be((login_page)))
+
+    # Заполнение поля Email на странице входа
+    driver.find_element(By.CSS_SELECTOR, email).send_keys("Denis_kvartych_15_333@yandex.ru")
```
можно лучше - тестовые данные тоже много раз повторяются, лучше было бы вынести в отдельный файл, в след спринтах пожалуйста выноси

irisqul commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,125 @@
+    WebDriverWait(driver, 5).until(expected_conditions.url_to_be((main_page)))
+
+    # Проверяем, что после авторизации произошел перевод на главную страницу и отображается кнопка "Оформить заказ"
+    assert driver.current_url == main_page and driver.find_element(By.CLASS_NAME, order_button).text == "Оформить заказ"
```
Можно лучше: мы проверяем не текст, а наличие элемента. Лучше перенести текст в селектор, это сделает локаторы надежнее, а в ассерте проверять присутствие элемента через isDisplayed

irisqul commented on urls.py:
```diff
@@ -0,0 +1,6 @@
+
+main_page = "https://stellarburgers.nomoreparties.site/"
+profile_page = "https://stellarburgers.nomoreparties.site/account/profile"
```
Можно лучше: Код станет более поддерживаемым, если вынести адрес сервера в отдельную переменную, а в переменных эндпоинтов оставить только их уникальный путь. Таким бразом, если нужно будет тестировать не продакшн или если поменяется url будет достаточно обновить одну переменную

AgonianV committed:
Добавлено закрытие страницы браузера в фикстуру.

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,17 @@
+    try:
+        yield driver
+    finally:
+        driver.quit()
```
нужно так:
    driver = webdriver.Chrome()
    yield driver
    driver.quit()

AgonianV committed:
Убрал из фикстуры try и finally.
